platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/roster/RosterUtils.java

```diff
@@ -338,4 +338,29 @@ public static AddressBook buildAddressBook(@NonNull final Roster roster) {
 
         return addressBook;
     }
+
+    /**
+     * Verifies that all nodes are the same between the two specified rosters.
+     *
+     * @param roster1 the first roster to compare
+     * @param roster2 the second roster to compare
+     * @throws IllegalStateException if the rosters do not match
+     */
+    public static void verifyReconnectRosters(@NonNull final Roster roster1, @NonNull final Roster roster2)
+            throws IllegalStateException {
+        final int numNodes = roster1.rosterEntries().size();
+
+        if (numNodes != roster2.rosterEntries().size()) {
+            throw new IllegalStateException("Rosters have different number of nodes");
+        }
+
+        for (int i = 0; i < numNodes; ++i) {
+            final RosterEntry roster1Entry = roster1.rosterEntries().get(i);
+            final RosterEntry roster2Entry = roster2.rosterEntries().get(i);
+
+            if (!roster1Entry.equals(roster2Entry)) {
+                throw new IllegalStateException("Rosters do not contain the nodes");
```

Comment:
Minor: it may be helpful to include some specific diagnostic information (e.g. the expected/found values) into the exception messages. Otherwise, it may be very difficult to debug when these exceptions do get thrown.

Reply:
That's true. I was trying to mirror what was done in the original AB verifier, which itself was lacking verbose information. In general, is it preferred to put details in the exception message or is it preferred to have short exception messages but also include a log statement along with the exception that contains more details?

Reply:
I would put sufficient detail in an exception that someone can know what the problem is just from seeing the exception.    A DevOps person will copy-paste an exception to get help.   Having to fish through a log can take longer, especially if they are on remote systems that we don't have access to.